lib/fdotypes.c

```diff
@@ -5490,20 +5490,214 @@ bool fdo_compare_rv_lists(fdo_rendezvous_list_t *rv_list1,
 			  fdo_rendezvous_list_t *rv_list2)
 {
 	bool retval = false;
+	int rv_directive_index = 0;
+	int rv_instr_index = 0;
+	fdo_rendezvous_list_t *rv_list1_traverse = rv_list1;
+	fdo_rendezvous_list_t *rv_list2_traverse = rv_list2;
 
-	if (!rv_list1 || !rv_list2) {
-		LOG(LOG_ERROR, "Null arguments!\n");
+	if (!rv_list1_traverse || !rv_list2_traverse) {
+		LOG(LOG_ERROR, "Received NULL arguments\n");
 		goto end;
 	}
 
-	// FIXME: rv_lists are very unlikely to change, being static scenario,
-	// therefore detailed check on rv_list is skipped.
+	if (rv_list1_traverse->num_rv_directives != rv_list2_traverse->num_rv_directives) {
+		LOG(LOG_ERROR, "Number of RendezvousDirective(s) do not match\n");
+		goto end;
+	}
+	while (rv_directive_index < rv_list1_traverse->num_rv_directives &&
+		rv_directive_index < rv_list2_traverse->num_rv_directives) {
+		fdo_rendezvous_directive_t *directive1 =
+			fdo_rendezvous_directive_get(rv_list1_traverse, rv_directive_index);
+		fdo_rendezvous_directive_t *directive2 =
+			fdo_rendezvous_directive_get(rv_list2_traverse, rv_directive_index);
+		if (!directive1 || !directive2) {
+			LOG(LOG_ERROR, "One of the RendezvousDirective(s) is empty\n");
+			goto end;
+		}
+		rv_instr_index = 0;
+		while (rv_instr_index < directive1->num_entries &&
```

Comment:
Does the value of num_entries need to match between directive1 and directive2?

Reply:
Yes. Alternatively, we could compare the num_entries once outside, and keep only 1 condition in the while-loop. But the end-result would be the same